wfe2/wfe_test.go

```diff
@@ -293,6 +293,28 @@ var testKeyPolicy = goodkey.KeyPolicy{
 	AllowECDSANISTP384: true,
 }
 
+// inMemNonceService implements noncepb.NonceServiceClient for tests.
+type inMemNonceService struct {
+	*nonce.NonceService
+}
+
+// Nonce implements proto.NonceServiceClient
+func (imns *inMemNonceService) Nonce(ctx context.Context, in *emptypb.Empty, opts ...grpc.CallOption) (*noncepb.NonceMessage, error) {
+	n, err := imns.NonceService.Nonce()
+	if err != nil {
+		return nil, err
+	}
+	return &noncepb.NonceMessage{Nonce: n}, nil
```

Comment:
I was surprised to see this doing actual work, instead of just `return imns.NonceService.Nonce(ctx, in)`. But then I realized that the //nonce package doesn't actually implement the nonce gRPC interface, despite containing nonce.proto -- actually implementing the interface is left to //cmd/nonce-service/main.go!

I'd use this as an opportunity to move that implementation into the nonce package itself, and then avoid having to duplicate that implementation in this inmem wrapper.